### ISSUE TITLE
Accessibility feedback for Project Kickoff for Left Nav

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#7027]
**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

### Must
1. Is there a mobile design for this? (Will update feedback when added) - comp added 03/20/20
2. Use semantic markup in development
3. If the `<nav>` element is used, provide an `aria-label` that distinguishes this nav from the main nav and the breadcrumbs nav, so screen readers can distinguish between them.

### Should
1. Should not use ALL CAPS for the section headings:
    1. Some screen readers read capitalized text letter-by-letter. This occasionally even occurs when the HTML is in sentence-case and the CSS forces the capitalization. VoiceOver is especially persnickety about this.
    1. "Text in all capital letters is more difficult to read for most people, with and without disabilities." Source: [Accessibility Requirements for People with Low Vision, Section 3.3.4](http://w3c.github.io/low-vision-a11y-tf/requirements.html#capitalization) All capital letters are harder to read (for everyone, but especially people with reading disabilities). The use of All Caps can reduce the readability of your text. Usability expert Jakob Nielsen has found that reading on screen can be around 25% slower than reading from paper, and reading All Caps can be a further 10% slower. It can be used for glance type uses, but it will still be an issue for screen reader users and people with dyslexia.
    1. For people with dyslexia, ALL CAPS text may have no difference in shape: all the words are just one big rectangle. Sentence capitals have differentiated shapes based on the letters used. When producing information to be read by others, it is important to remember that up to 10% of your readers may have dyslexia. Dyslexia friendly text will have improved readability and better visual impact for all readers, but especially those with dyslexia.
1. If ALL CAPS are used, use `aria-label="Sentence case words"` so all screen readers will read the words rather than spelling out letters. Please validate with all screen readers available to dev.
1. How does the focus yellow outline pair with the outline of the :hover and :active states? 
1. Should include a notification of on-state for screen readers. 

```diff
! The VSA accessibility specialist offers to spec focus management 
! and other accessibility considerations in the design, if that helps.
```

### Consider

**Mobile** The "on" state has a color contrast ratio of 1.06:1 of the light grey background of the line item and white background area. Looking ahead to WCAG 2.1, this will be non-compliant. While noting the #2d71b6 small bar on the left, given the small screen size it may not be sufficiently prominent. I'm having a hard time seeing it on my monitor next to the window with daylight, for example. 

### Must
1. Include markup for properly naming this `<nav>` element for screen readers. We already have two on most pages, so having a name that is descriptive and can be read aloud will be important. Example snippet:

    ```html
    <nav aria-label="Services and locations">...</nav>
    ```
2. Make sure the divider lines are either hidden programmatically or are borders created with CSS. Could be `<hr aria-hidden="true" />` or `border-bottom: 1px solid #ccc`

### Should
1. Each section should be a properly nested heading like `<h3>Services and Locations</h3>`
2. Should define a focus state. Do we use the standard yellow focus halo, or a derivative thereof?
3. Announce "on state" using `aria-current="page"` attribute on the current `<a>` tag

### Consider
1. A keyboard-focus only "skip link" to jump from heading to heading. There's a number of links here, especially when we add child and grand-child links. See [Bypass Blocks: Understanding SC 2.4.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-skip.html)

## Definition of Done
1. Acknowledge feedback
2. Document decisions made in comment on this ticket
3. Close ticket

**Comment by mmiddaugh** [comment]
Acknowledging feedback, @jenstrickland 
@rtwell to review

**Comment by rtwell** [comment]
@jenstrickland 

under the *Should* section for accessibility feedback, fo2 number 2: no, I don't think we want a different focus state treatment than the rest of va.gov.

**Comment by rtwell** [comment]
@jenstrickland + @1Copenut should I be commenting on each line item? I'm not sure what you're looking for from me/how to get to next steps. Thank you!

**Comment by jenstrickland** [comment]
@rtwell Would you go through the design-relevant items in the feedback and document decisions made, please? It's okay to do it all in one comment, which what other designers have done. 

Some of the feedback is developer content, which you don't have to acknowledge. I'm working on defining a process so that goes to the developers.

**Comment by jenstrickland** [comment]
@rtwell Would you please mockup how the focus outline would pair with the design, so the developers know what it should look like?

**Comment by rtwell** [comment]
# design relevant feedback

## VSP Accessibility feedback

### Must
1. attached to epic
2. N/A
3. N/A

### Should
1. While I agree all caps isnt the easiest to read, we're using them here. If there is some kind of CSS magic that will ease usability from the screen reader perspective, that would be ideal
2. The focus state should be consistent with the rest of VA.gov. Personally, I don't see why a mock up is needed as it should follow suit, but maybe we want to be explicit. Here they are:
![image](https://user-images.githubusercontent.com/12564977/77365138-f9314b00-6d2b-11ea-8ecb-9c92b852fa13.png)
3. N/A

### Consider
1. I think this is N/A, but I love this idea!

## VSA Accessibility feedback

### Must
1. N/A
2. N/A

### Should
1. N/A
2. The focus state should be consistent with the rest of VA.gov. Personally, I don't see why a mock up is needed as it should follow suit, but maybe we want to be explicit. Here they are:
![image](https://user-images.githubusercontent.com/12564977/77365138-f9314b00-6d2b-11ea-8ecb-9c92b852fa13.png)
3. N/A

### Consider
1. N/A

**Comment by 1Copenut** [comment]
@rtwell Thank you for attaching the focus state. That's the big one for definition of done for me. The others were code recommendations that I'm still experimenting with how to communicate to teams. If there's a way to attach them as notes to the epic, or you have another suggestion, let me know.

**Comment by jenstrickland** [comment]
@rtwell 

- I asked for a mockup of the focus state as I worried it wouldn't be clear to the dev where the yellow outline should be. A dev recently implemented a focus outline so it was cut off on one side. I wasn't clear if this design was going to abut the border, overlap, or be spaced a distance from it. For the sake of development, I hoped to be clear upfront.

- We ought to hop on a call sometime to hear how screen readers read these pages. The ALL CAPS is tough. Developers can code from Sentence case HTML, use CSS to render the ALL CAPS, but some screen readers still read the individual letters.

**Comment by rtwell** [comment]
@1Copenut @jenstrickland regarding focus mocks—noted. it sounds like we should work on our design review process to catch things like erroneous focus state, maybe? I'm all for providing as much detail as needed, but pointing to a component on design.va.gov and saying "insert this component/functionality/whatever here" should be enough for any developer to execute. 

to your second point, @jenstrickland, I appreciate you advocating for this, but I am making the  decision to use the caps.

**Comment by jenstrickland** [comment]
@rtwell 👍 I'll add some developer notes to provide ARIA spackle to over-ride the all caps, and see if that improves the screen reader experience.

**Comment by mmiddaugh** [comment]
@jenstrickland can this be closed?
